src/main/java/com/hibernate/cinema/controller/CinemaHallController.java

```diff
@@ -0,0 +1,40 @@
+package com.hibernate.cinema.controller;
+
+import com.hibernate.cinema.model.dto.CinemaHallRequestDto;
+import com.hibernate.cinema.model.dto.CinemaHallResponseDto;
+import com.hibernate.cinema.service.CinemaHallService;
+import com.hibernate.cinema.service.mapper.CinemaHallMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private final CinemaHallService cinemaHallService;
+    private final CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping
+    public void addCinemaHall(@RequestBody CinemaHallRequestDto cinemaHall) {
+        cinemaHallService.add(cinemaHallMapper.castDtoToCinemaHall(cinemaHall));
```

Comment:
```suggestion
        cinemaHallService.add(cinemaHallMapper.mapDtoToCinemaHall(cinemaHall));
```
to avoid confusing with type casting in java